src/app/pages/public/DetailLifePage/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState } from 'react';
+import { Card } from 'antd';
+import { useParams, useNavigate } from 'react-router-dom';
+import BasicLayout from '../../../layouts/BasicLayout';
+import { useGetLifeQuery } from '../../../api';
+
+const DetailLifePage = () => {
+    const { id } = useParams();
+    const navigate = useNavigate();
+
+    const { data, loading, error } = useGetLifeQuery({ variables: { id } });
+
+    if(error) {
+        navigate('/404');
```

Comment:
cannot call navigate in rendering, in fact it would be better rendering a 404 components here